### PR TITLE
Add logic to switch with "under_maintenance" flag.

### PR DESCRIPTION
**Pull request by dpoltavets**
API changes: added `evacuate` flag to set/unset Switch under_maintenance API

Closes: #1703.

Depends on: #1883.

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/1961)

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java [hunk 1]
```diff
@@ -73,4 +74,14 @@ private void processDeactivateSwitch(Switch sw) {
         daoSwitch.setStatus(SwitchStatus.INACTIVE);
         switchRepository.createOrUpdate(daoSwitch);
     }
+
+    /**
+     * Return switch "Under maintenance" flag.
+     * @param switchId switch id.
+     */
+    public boolean switchIsUnderMaintenance(SwitchId switchId) {
```
Comment (IlyaMoiseev) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java]: may be simplified to
```
return switchRepository.findById(switchId)
                .map(Switch::isUnderMaintenance)
                .orElse(false);
```

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 2]
```diff
@@ -57,4 +61,27 @@ public FlowOperationsService(RepositoryFactory repositoryFactory) {
...
         return flowRepository.findAllFlowPairsWithSegment(srcSwitchId, srcPort, dstSwitchId, dstPort);
     }
+
+    /**
+     * Return flows for a switch.
+     *
+     * @param srcSwitchId switch id.
+     * @return all flows for a switch.
+     */
+    public Set<String> getFlowIdsForSwitch(SwitchId srcSwitchId) {
```
Comment (IlyaMoiseev) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: Looks like new DB query candidate

File: services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java (hunk 1 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java]: Thank you. You are right.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java (hunk 2 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: You are right. Fixed.

File: services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Switch.java [hunk 3]
```diff
@@ -68,6 +68,7 @@
 
     private String description;
 
+    @Property(name = "under_maintenance")
```
Comment (timofei-durakov) [inline services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Switch.java]: should't this part be in the previous commit?

Comment (dpoltavets) [reply services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Switch.java]: Yes, you are right. I forgot to write this annotation in the previous commit. But we plan to merge this branch in develop as soon as possible and tests will slow it down. Anyway, these commits need to be merged together.

**Comment by dpoltavets** [comment]
___
*[services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/request/UpdateSwitchUnderMaintenanceRequest.java, line 42 at r1](https://reviewable.io/reviews/telstra/open-kilda/1961#-LXXlynRD1t0_7C5n-p7:-LXcEOUt9Wk-hqG-kquq:bt2vr1b) ([raw file](https://github.com/telstra/open-kilda/blob/edd585fefcda87d578f19017a188996a33669a32/services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/request/UpdateSwitchUnderMaintenanceRequest.java#L42)):*
<details><summary><i>Previously, niksv (Sergey Nikitin) wrote…</i></summary><blockquote>

minor: 'value is useless'
</blockquote></details>

Fixed.

File: services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Switch.java (hunk 3 above)
Comment (dpoltavets) [reply services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Switch.java]: Fixed.

**Pull request merged**